src/Sarif.Viewer.VisualStudio.Core/SarifViewerPackage.cs

```diff
@@ -171,9 +171,10 @@ private async Task InitializeResultSourceHostAsync()
             string solutionPath = GetSolutionDirectoryPath();
             if (!string.IsNullOrWhiteSpace(solutionPath) && SarifViewerOption.Instance.IsGitHubAdvancedSecurityEnabled)
             {
-                this.resultSourceHost = new ResultSourceHost(solutionPath, this);
+                var resultSourceFactory = new ResultSourceFactory(solutionPath, this);
+                this.resultSourceHost = new ResultSourceHost();
                 this.resultSourceHost.ResultsUpdated += this.ResultSourceHost_ResultsUpdated;
-                await this.resultSourceHost.RequestAnalysisResultsAsync();
+                await this.resultSourceHost.RequestAnalysisResultsAsync(resultSourceFactory);
```

Comment:
>RequestAnalysisResultsAsync [](http://example.com/codeflow?start=44&length=27)

is it possible this call or the initializing of ResultSourceFactory/Host throw exception? should we catch here? Just want to make sure if any unexpected error happens inside ResultsourceFactory/Host, rest of the viewer extension can still run.